src/login/logind-dbus.c

```diff
@@ -1509,8 +1509,10 @@ static int method_set_user_linger(sd_bus_message *message, void *userdata, sd_bu
                         return -errno;
 
                 u = hashmap_get(m->users, UID_TO_PTR(uid));
-                if (u)
+                if (u) {
+                        u->gc_mode = USER_GC_BY_PIN;
```

Comment:
can you add a comment here, maybe something like: `/* Make sure that disabling lingering will terminate the user tracking if no sessions pin it. */`  or so?